test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/DistributedCacheTagHelperTest.cs

```diff
@@ -0,0 +1,926 @@
+        {
+            // Arrange
+            var name = "some-name";
+            var expected = GetHashedBytes("DistributedCacheTagHelper||some-name||VaryBy||custom-value||" +
+                                          "VaryByHeader(content-type||text/html)||VaryByUser||someuser");
+            var tagHelperContext = GetTagHelperContext();
+            var memoryCache = new MemoryCache(new MemoryCacheOptions());
+            var cacheTagHelper = new DistributedCacheTagHelper(
+                memoryCache,
+                Mock.Of<IDistributedCacheTagHelperStorage>(), 
+                Mock.Of<IDistributedCacheTagHelperFormatter>(), 
+                new HtmlTestEncoder())
+            {
+                ViewContext = GetViewContext(),
+                VaryByUser = true,
+                VaryByHeader = "content-type",
+                VaryBy = "custom-value",
+                Name = name
+            };
+            cacheTagHelper.ViewContext.HttpContext.Request.Headers["Content-Type"] = "text/html";
+            var identity = new ClaimsIdentity(new[] { new Claim(ClaimsIdentity.DefaultNameClaimType, "someuser") });
+            cacheTagHelper.ViewContext.HttpContext.User = new ClaimsPrincipal(identity);
+
+            // Act
+            var key = cacheTagHelper.GenerateKey(tagHelperContext);
+
+            // Assert
+            Assert.Equal(expected, key);
+        }
+
+        [Fact]
+        public async Task ProcessAsync_DoesNotCache_IfDisabled()
+        {
+            // Arrange
+            var childContent = "original-child-content";
+            var memoryCache = new MemoryCache(new MemoryCacheOptions());
+            var storage = new Mock<IDistributedCacheTagHelperStorage>();
+            storage.CallBase = true;
+            var value = Encoding.UTF8.GetBytes("ok");
+            storage.Setup(c => c.SetAsync(
+                /*key*/ It.IsAny<string>(),
+                /*value*/ value,
+                /*options*/ It.IsAny<DistributedCacheEntryOptions>()))
+                .Verifiable();
+            storage.Setup(c => c.GetAsync(It.IsAny<string>()))
+                .Returns(Task.FromResult(value));
+            var tagHelperContext = GetTagHelperContext();
+            var tagHelperOutput = GetTagHelperOutput(
+                attributes: new TagHelperAttributeList(),
+                childContent: childContent);
+            var cacheTagHelper = new DistributedCacheTagHelper(
+                memoryCache,
+                storage.Object, 
+                Mock.Of<IDistributedCacheTagHelperFormatter>(),
+                new HtmlTestEncoder())
+            {
+                ViewContext = GetViewContext(),
+                Enabled = false
+            };
+
+            // Act
+            await cacheTagHelper.ProcessAsync(tagHelperContext, tagHelperOutput);
+
+            // Assert
+            Assert.Equal(childContent, tagHelperOutput.Content.GetContent());
+            storage.Verify(c => c.SetAsync(
```

Comment:
Didn't see the `Times.Never`. In which case, remove the `Setup \ Verifiable` call from Line 318